test/spiffe_id/test_spiffe_id.py

```diff
@@ -0,0 +1,184 @@
+    ],
+)
+def test_parse_spiffe_id_valid_uri(spiffe_id_str, expected_trust_domain, expected_path):
+    spiffe_id = SpiffeId.parse(spiffe_id_str)
+    assert spiffe_id.trust_domain() == expected_trust_domain
+    assert spiffe_id.path() == expected_path
+
+
+@pytest.mark.parametrize(
+    'spiffe_id_str,expected',
+    [
+        ('', 'SPIFFE ID cannot be empty.'),
+        (None, 'SPIFFE ID cannot be empty.'),
+        ('192.168.2.2:6688', "'192.168.2.2:6688' is not a valid 'URI'."),
+        (
+            'http://domain.test/path/element',
+            'SPIFFE ID: invalid scheme: expected spiffe.',
+        ),
+        ('spiffe:///path/element', 'SPIFFE ID: trust domain cannot be empty.'),
+        (
+            'spiffe://domain.test/path/element?query=1',
+            'SPIFFE ID: query is not allowed.',
+        ),
+        (
+            'spiffe://domain.test/path/element#fragment-1',
+            'SPIFFE ID: fragment is not allowed.',
+        ),
+        ('spiffe://domain.test:8080/path/element', 'SPIFFE ID: port is not allowed.'),
+        (
+            'spiffe://user:password@test.org/path/element',
+            'SPIFFE ID: user info is not allowed.',
+        ),
+        ('spiffe:path/element', 'SPIFFE ID: trust domain cannot be empty.'),
+        ('spiffe:/path/element', 'SPIFFE ID: trust domain cannot be empty.'),
+        (
+            'spiffe://domain.test/path/elem%5uent',
+            "'spiffe://domain.test/path/elem%5uent' is not a valid 'URI'.",
+        ),
+    ],
+)
+def test_parse_spiffe_id_from_invalid_uri_string(spiffe_id_str, expected):
+    with pytest.raises(ValueError) as exception:
+        SpiffeId.parse(spiffe_id_str)
+
+    assert str(exception.value) == expected
+
+
+def test_not_equal_spiffe_ids():
+    trust_domain = TrustDomain('domain.test')
+    spiffeid_1 = SpiffeId.of(trust_domain, 'path1')
+    spiffeid_2 = SpiffeId.of(trust_domain, 'path2')
+    assert spiffeid_1 != spiffeid_2
+
+
+def test_equal_spiffe_id():
+    trust_domain = TrustDomain('domain.test')
+    spiffeid_1 = SpiffeId.of(trust_domain, 'path1')  # path1 is normalized as /path1
+    spiffeid_2 = SpiffeId.of(trust_domain, '/path1')
+    assert spiffeid_1 == spiffeid_2
+
+
+def test_equal_spiffe_id_with_multiple_paths():
+    trust_domain = TrustDomain('example.org')
+    spiffeid_1 = SpiffeId.of(trust_domain, ['PATH1', 'PATH2'])
+    spiffeid_2 = SpiffeId.of(trust_domain, ['/PATH1', '/PATH2'])
+    assert spiffeid_1 == spiffeid_2
+
+
+def test_trust_domain_none():
+    with pytest.raises(ValueError) as exception:
+        SpiffeId.of(None, 'path')
+
+    assert str(exception.value) == 'SPIFFE ID: trust domain cannot be empty.'
+
+
+def test_of_empty_trust_domain():
+    with pytest.raises(ValueError) as exception:
+        SpiffeId.of('', 'path')
+
+    assert (
+        str(exception.value)
+        == 'SPIFFE ID: trust_domain argument must be a TrustDomain instance.'
+    )
+
+
+def test_exceeds_maximum_length():
+    path = "a" * 2028
```

Comment:
nit: there are some places where double quotes are used instead of single quotes. Let's use only single quotes to be consistent.